newless.js

```diff
@@ -33,6 +33,10 @@
 
       return Function("constructor, args, target",
         "'use strict';" +
+
+        "if (arguments.length === 3 && typeof target !== 'function')" +
+          "throw new TypeError(''+target+' is not a constructor');" +
+
```

Comment:
I’m not sure this is particularly valuable. In the abstract, it makes sense because `Reflect.construct` is strict about what it allows, but practically speaking, this function is only available privately, where we don’t really need to worry about validating these values. Is there a case where this validation is going to help us?

And if we did want this, we should be checking `arguments.length >= 3`, not `=== 3` if we want to match spec & implementations. We should also be type checking the first and second arguments.

Please also keep whitespace around operators, like ` + `.

Same for the identical block you have in the alternate, non-class-supporting branch below.